### PR TITLE
Nominatim 4.5

### LINKED ISSUE #584: Nominatim 4.5
**Is your feature request related to a problem? Please describe.**
Nominatim 4.5 was released a month ago and seems to be quite a large step toward the v5 goals. 

In 4.5 there is response customization functionality related to the python frontend replacement that I would like to access without having to revert to an installation on the host. Specifically this was suggested by Sarah Hoffman for being able to return building level (18) AND street level (16) data in one request.

**Describe the solution you'd like**
Has a release of mediagis/nominatim-docker 4.5 been considered / perhaps in progress?

**Describe alternatives you've considered**
There are no real 'good' alternatives available. Double calls to the DB would be problematic.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by leonardehrenfried**
Nominatim 4.5 is moving to Python and the PHP part is now deprecated.

For that reason I removed PHP and apache. 

I also upgraded to Ubuntu 24.04 which also uses Postgres 16.

Closes #584.

### CODE REVIEW

**Comment by leonardehrenfried** [comment]
> The CI configuration has been updated to consider a wider geographical region - shifting the focus from Rutland to England which could expand the relevance of the data analysis.

Again, this is just great! :laughing:

**Comment by leonardehrenfried** [comment]
OTOH, this is quite good:

> Three new scripts have been provided for managing various aspects of the application lifecycle - init.sh, start.sh, and startpostgres.sh. They broadly handle data file downloads, initialization of PostgreSQL database, performing initial data import, setting up the application, and monitoring application operations.

File: 4.5/contrib/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+version: "3"
+
+services:
+    nominatim:
+        container_name: nominatim
+        image: mediagis/nominatim:4.4
```
Comment (mtmail) [inline 4.5/contrib/docker-compose.yml:6]: should be 4.5 here

**Comment by leonardehrenfried** [comment]
It looks like I encountered a bug in nominatim 4.5.0 when importing with `--reverse-only`: https://github.com/osm-search/Nominatim/issues/3583

**Review by philipkozeny — APPROVED** [review]
amazing work, thanks @leonardehrenfried !

**Pull request merged**